shell/src/main/java/org/apache/kafka/shell/MetadataNodeManager.java

```diff
@@ -318,6 +320,15 @@ private void handleCommitImpl(MetadataRecordType type, ApiMessage message)
                     node.create(record.key()).setContents(record.value() + "");
                 break;
             }
+            case PRODUCER_IDS_RECORD: {
+                ProducerIdsRecord record = (ProducerIdsRecord) message;
+                DirectoryNode producerIdNode = data.root.mkdirs("producerIds");
+                producerIdNode.create("broker").setContents(record.brokerId() + "");
```

Comment:
Maybe `assignedBrokerId`? As in, this was the brokerId that the block was assigned to. Also, perhaps we may as well add `assignedBrokerEpoch` as well? I guess we could even do it like `/lastProducerBlock/assignedBroker/{id,epoch}`

Reply:
Thank you for these suggestions, I think assignedBrokerId and assignedBrokerEpoch are enough since we don't use JSON format when parsing other metadata records.